src/parse.rs

```diff
@@ -1201,12 +1201,16 @@ pub(crate) struct CowIndex(usize);
 #[derive(Copy, Clone, PartialEq, Eq, Debug)]
 pub(crate) struct AlignmentIndex(usize);
 
+#[derive(Copy, Clone, PartialEq, Eq, Debug)]
+pub(crate) struct HeadingIndex(usize);
+
 #[derive(Clone)]
 pub(crate) struct Allocations<'a> {
     pub refdefs: RefDefs<'a>,
     links: Vec<(LinkType, CowStr<'a>, CowStr<'a>)>,
     cows: Vec<CowStr<'a>>,
     alignments: Vec<Vec<Alignment>>,
+    headings: Vec<(HeadingLevel, Option<CowStr<'a>>, Vec<CowStr<'a>>)>,
```

Comment:
This is a fine solution, but I'm just wondering whether it could pay off to store `HeadingLevel` inline and only store the id and classes in the `Allocations` struct, if they are applicable. Not sure if this makes sense, but maybe you've thought about it already.

Reply:
I saw `LinkType` (which is a simple enum, similar to `HeadingLevel`) being included to the links allocation, so followed to that style.
But yes, `HeadingLevel` can be put into `ItemBody::Heading` directly, and I think it is better since it does not require extra indirection.
I'll move `HeadingLevel` value from `Allocations` into `ItemBody::Heading`.